numpy/core/numeric.py

```diff
@@ -1457,6 +1461,93 @@ def rollaxis(a, axis, start=0):
     return a.transpose(axes)
 
 
+def _validate_axis(axis, ndim, argname):
+    try:
+        axis = [operator.index(axis)]
+    except TypeError:
+        axis = list(axis)
+    original_axis = axis
+    axis = [a + ndim if a < 0 else a for a in axis]
+    if not builtins.all(0 <= a < ndim for a in axis):
+        raise ValueError('%s=%s has elements outside valid bounds' %
+                         (argname, original_axis))
+    if len(set(axis)) != len(axis):
+        raise ValueError('repeated axis in %s=%s' % (argname, original_axis))
+    return axis
+
+
+def moveaxis(a, source, destination):
+    """
+    Move axes of an array to new positions.
```

Comment:
Just noticed this is still open, sorry for forgetting about it (dunno if it was me who reviewed it first). Mostly some thoughts left.
I think we might want to mention the word "view" here (and/or in the Result section). It should be somewhat obvious to most users (and seems also missing for transpose), but I think it is quite an important word in general.